src/modify.js

```diff
@@ -66,12 +101,122 @@ function rewriteAllFields(schema, configCallback, context) {
   });
 }
 
+function pickFields(originalSchema, pickConfig) {
+  if (!pickConfig) return originalSchema;
+
+  const fieldsToPick = pickConfig.fields;
+  const newSchema = {
+    properties: {},
+  };
+
+  Object.entries(originalSchema).forEach(([attrKey, attrValue]) => {
+    switch (attrKey) {
+      case 'properties':
+        // TODO — handle recursive nested fieldsets
+        fieldsToPick.forEach((fieldPath) => {
+          set(newSchema.properties, fieldPath, attrValue[fieldPath]);
+        });
+        break;
+      case 'x-jsf-order':
+      case 'required':
+        newSchema[attrKey] = attrValue.filter((fieldName) => fieldsToPick.includes(fieldName));
+        break;
+      case 'allOf': {
+        const newAllOf = [];
+        // remove conditional ("if, then, else") if it does not contain any reference to fieldsToPick
+        originalSchema.allOf.forEach((condition) => {
+          if (isConditionalReferencingAnyPickedField(condition, fieldsToPick)) {
+            newAllOf.push(condition);
+          }
+        });
+        newSchema[attrKey] = newAllOf;
+
+        break;
+      }
+      default:
+        newSchema[attrKey] = attrValue;
+    }
+  });
+
+  // Look for unpicked fields in the conditionals...
+  let missingFields = {};
+
+  newSchema.allOf.forEach((condition, ix) => {
+    const { if: ifCondition, then: thenCondition, else: elseCondition } = condition;
+
+    missingFields = {
+      ...missingFields,
```

Comment:
Oh yes we can! I simplified the code so much but forgot that part, thanks!

Reply:
Hmm, I looked again, actually we need to keep this one because it's a loop, otherwise it will ignore any findings from previous conditionals. I added a test to show you, if you delete this line you'll see it failing

Reply:
ah, apparently my brain circuitry forgot to deal to re-assignments 😅 we could use `Array.reduce` here then, but no need to change.